plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetastoreClosure.java

```diff
@@ -87,13 +88,39 @@ public Set<HiveColumnStatisticType> getSupportedColumnStatistics(Type type)
 
     public PartitionStatistics getTableStatistics(String databaseName, String tableName)
     {
-        return delegate.getTableStatistics(getExistingTable(databaseName, tableName));
+        return getTableStatistics(databaseName, tableName, Optional.empty());
+    }
+
+    public PartitionStatistics getTableStatistics(String databaseName, String tableName, Optional<Set<String>> columns)
+    {
+        Table table = getExistingTable(databaseName, tableName);
+        if (columns.isPresent()) {
+            Set<String> requestedColumnNames = columns.get();
+            List<Column> requestedColumns = table.getDataColumns().stream()
+                    .filter(column -> requestedColumnNames.contains(column.getName()))
+                    .collect(toImmutableList());
+            table = Table.builder(table).setDataColumns(requestedColumns).build();
```

Comment:
we're building fake `table` object, hoping the callee will trust the `table`, and not eg use knowledge about column list from some other place

that's an opaque design, something explicit would probably be better.
Why not pass `Set<String> requestedColumns` (or `Optionale` thereof`) down the call chain?

Reply:
> Why not pass Set<String> requestedColumns (or Optional thereof`) down the  call chain?

That's an excellent idea. It also applies to the `getPartitionStatistics`, right?